modules/channel-smooch/src/backend/client.ts

```diff
@@ -0,0 +1,128 @@
+import * as sdk from 'botpress/sdk'
+import Smooch from 'smooch-core'
+
+import { Config } from '../config'
+
+import { Clients } from './typings'
+
+const name = 'smooch.sendMessage'
+
+export class SmoochClient {
+  private smooch: any
+  private path: string
+
+  constructor(
+    private bp: typeof sdk,
+    private botId: string,
+    private config: Config,
+    private router: any,
+    private route: string
+  ) {}
+
+  async initialize() {
+    this.smooch = new Smooch({
+      keyId: this.config.keyId,
+      secret: this.config.secret,
+      scope: 'app'
+    })
+
+    const url = (await this.router.getPublicPath()) + this.route
+    this.path = url.replace('BOT_ID', this.botId)
+
+    await this.smooch.webhooks.create({
+      target: this.path,
+      triggers: ['message:appUser']
+    })
+  }
+
+  async delete() {
+    const { webhooks } = await this.smooch.webhooks.list()
+    for (const hook of webhooks) {
+      if (hook.target == this.path) {
+        await this.smooch.webhooks.delete(hook._id)
```

Comment:
This looks like a private property, is it the "correct" way to delete it ?

Reply:
As you suggested this function will be renamed removeWebhook. I don't think there's any fancier way to remove them. We simply get a list of the existing webhooks and delete those that target our url ("path" or its new name "webhookUrl"). list() and delete() are API calls to smooch.